### ISSUE TITLE
Segfault with `BetaCoalescent` and ploidy greater than diploid

### ISSUE THREAD

**Issue by nspope** [tskit-dev/msprime#2307]
The following snippet gives me a segfault:

```python
import msprime
print(msprime.__version__) # 1.3.2
msprime.sim_ancestry(
    samples=10,
    recombination_rate=1e-8,
    sequence_length=1e6,
    population_size=1e4,
    ploidy=3,
    random_seed=1024,
    model=msprime.BetaCoalescent(alpha=1.5),
)
```

**Comment by andrewkern** [comment]
fwiw I can recreate a similar error on msprime 1.3.1 on my mac laptop, although it's telling me `abort` rather than segfault

**Comment by jeromekelleher** [comment]
Confirmed also on Linux, v 1.3.1. Investigating.

**Comment by jeromekelleher** [comment]
#2308 fixes the segfault. I'd need some help in figuring out whether it's actually doing the right thing, though.

**Comment by JereKoskela** [comment]
Making `num_parental_copies` the number of buckets is the right thing to do.

Statistical validation is not totally straightforward because this is the only implementation for ploidy > 2 of which I'm aware. We could moments of TMRCAs of n = 2, for which the expectation and variance are easy to compute for various ploidies, say 1, ..., 6.

**Comment by jeromekelleher** [comment]
> Statistical validation is not totally straightforward because this is the only implementation for ploidy > 2 of which I'm aware. We could estimate moments of TMRCAs of n = 2, for which the expectation and variance are easy to compute for various ploidies, say 1, ..., 6.

That sounds great - would you be able to pick this up if I merge #2308?

I'll push out a bugfix release in the next couple of days, as we don't really want segfault bugs lying around.

**Comment by JereKoskela** [comment]
Sure, I'll put something together.

**Comment by jeromekelleher** [comment]
Great, thanks a lot. I'll push out the bugfix once we're reasonably sure that the fix works.

**Comment by jeromekelleher** [comment]
Closed in #2310 and #2308. Great work everyone!
